src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs

```diff
@@ -10,44 +10,46 @@ internal partial struct FileStatus
     {
         private const int NanosecondsPerTick = 100;
 
-        // The last cached lstat information about the file
+        private const int InitializedExists = -2;    // entry exists.
+        private const int InitializedNotExists = -1; // no entry exists.
+        private const int Uninitialized = 0;         // uninitialized, '0' to make default(FileStatus) uninitialized.
+
+        // Tracks the initialization state.
+        // < 0 : initialized succesfully. Value is InitializedExists or InitializedNotExists.
+        //   0 : uninitialized.
+        // > 0 : initialized with error. Value is raw errno.
+        private int _initialized;
+
+        // The last cached lstat information about the file.
+        // Must only be used after calling EnsureCachesInitialized and checking FileExists is true.
         private Interop.Sys.FileStatus _fileCache;
 
-        // -1: if the file cache isn't initialized - Refresh should always change this value
-        //  0: if the file cache was initialized with no errors
-        // Positive number: the error code returned by the lstat call
-        private int _initializedFileCache;
-
-        // We track intent of creation to know whether or not we want to (1) create a
-        // DirectoryInfo around this status struct or (2) actually are part of a DirectoryInfo.
-        // Set to true during initialization when the DirectoryEntry's INodeType describes a directory
-        internal bool InitiallyDirectory { get; set; }
-
-        // Is a directory as of the last refresh
-        // Its value can come from either the main path or the symbolic link path
+        // Is a directory as of the last refresh.
+        // Its value can come from either the main path or the symbolic link target.
+        // Must only be used after calling EnsureCachesInitialized.
         private bool _isDirectory;
 
-        // Exists as of the last refresh
-        private bool _exists;
+        private bool FileExists => _initialized == InitializedExists;
```

Comment:
This replaces `_exists`.
It may be the separate `bool` has a measurable performance gain over this comparison. If that is so, it can be added back.